download_weather_data.py

```diff
@@ -31,7 +31,7 @@ def http_error_default(self, url, fp, errcode, errmsg, headers):
 
 def fetch_noaa_data(start, end):
 
-    path = '/spindisk/weather_data/gfs_anl/'
+    path = './weather_data/'  # was '/spindisk/weather_data/gfs_anl/'
```

Comment:
I would recommend leaving this in a global directory. The reason is that usually you will want this data to be on a separate disk